### PR TITLE
Thread viewedGroup into PageComponents; drop redundant viewedGroupId

### LINKED ISSUE #488: Thread viewedGroup into PageComponents; drop redundant viewedGroupId
## Context

Part of the "Group slugs in URLs" effort (#472), step 3 of 3 — depends on #487. This is where `viewedGroupId` finally disappears from the `PageComponent` boundary and from the `/group` wrapper layer's calls.

## Scope

- `app/components/layout/BootstrapPageData.tsx`: `PageComponent`'s prop type changes `viewedGroupId: number` → `viewedGroup: { id: number; slug: string | null }`; `LoadWithData`'s render passes `viewedGroup={viewedGroup}` instead of `viewedGroupId={viewedGroupId}`. Update the mock's `PageComponent` call the same way (`viewedGroup={{ id: 1, slug: 'alpha' }}` instead of `viewedGroupId={1}`).
- Update the `PageComponent` implementations that actually destructure `viewedGroupId` today (confirmed by grep — the rest receive it structurally but never read it, so they need no change):
  - Inline in their route file: `HomePageContent` (`app/(routes)/page.tsx`), `PayOffPageContent` (`effort/page.tsx`), `ListAllSessions` (`sessions/page.tsx`), `BirdSummary` (`bird/[ring]/page.tsx`), `HighlightsPageContent` (`highlights/page.tsx`)
  - Imported components: `SpPage.tsx`, `SppStatsTable.tsx`, `ControlsPage.tsx`, `RingSequencesPage.tsx`
  - Each swaps its `viewedGroupId: number` prop for `viewedGroup: { id: number; slug: string | null }` and uses `viewedGroup.id` wherever it previously used `viewedGroupId` (including when passing the plain numeric id further down to its own children, e.g. `SpPage` → `SpStats`/`SpYearComparisonTab`/etc., `ListAllSessions` → `SessionHistoryCalendar`). Those grandchildren keep receiving a plain `viewedGroupId: number` as before — this ticket does not cascade the object shape past the `PageComponent` boundary.
  - No change needed: `ListTicks`, `ListMistakes`, `ListResightings`, `ListNotableRetraps`, `ListPulliEncounters`, `BirdSearchResults` (none destructure `viewedGroupId`).
- Drop the now-redundant `viewedGroupId={viewedGroupId}` prop from the 13 `/group/[groupId]/**` wrapper files added in #486 — they pass only `viewedGroup` from here on.
- Correspondingly drop `viewedGroupId?: number` entirely from the 15 `app/(routes)/*/page.tsx` prop types — `viewedGroup` becomes the sole override input.

## Out of scope

- Route rename (see ticket 4, following this one).
- Any deeper prop-drilling refactor of the tab/chart components downstream of the 9 `PageComponent`s above.

## Files touched

- `app/components/layout/BootstrapPageData.tsx`, its mock
- 9 `PageComponent` implementation files
- 13 `/group/[groupId]/**` wrapper files (drop one prop each)
- 15 `app/(routes)/*/page.tsx` files (drop `viewedGroupId` from prop type)

## Commits

1. `BootstrapPageData`: swap `PageComponent`'s `viewedGroupId` output for `viewedGroup`; update mock
2. Update the 9 `PageComponent` implementations to consume `viewedGroup`
3. Drop redundant `viewedGroupId` from the 13 wrapper files and the 15 route-page prop types

## Tests (USE)

- `BootstrapPageData`: `PageComponent` receives `viewedGroup` (not `viewedGroupId`) — update existing render-assertion tests.
- Per updated `PageComponent` (9 files): existing tests updated to render with `viewedGroup={{ id, slug }}` instead of `viewedGroupId={id}` — Usual (renders correctly with the object), plus any existing edge-case tests (e.g. `SppStatsTable`'s year-filter tests) continue to pass unchanged in behaviour, only the prop shape changes.
- Regression: full `npm run test:nowatch` run — this ticket touches ~40 files, so the fastest way to catch a missed callsite is the full suite plus `npm run type:check` (TypeScript will flag any component still expecting `viewedGroupId` once the shared type changes).

### PR DESCRIPTION
**Pull request by wheresrhys**
## What this covers

Step 3 of 3 in the "Group slugs in URLs" effort (#472), building directly on #486 (`lib/group-slug.ts` resolver + `ViewedGroup` type) and #487 (`BootstrapPageData` accepts a `viewedGroup` input prop), both already merged.

This is where `viewedGroupId` disappears from the `PageComponent` boundary and from the `/group` wrapper layer's calls:

- **`BootstrapPageData`**: `PageComponent`'s prop type changes `viewedGroupId: number` → `viewedGroup: ViewedGroup`; `LoadWithData` now narrows on the resolved `viewedGroup` object directly (so the non-null object flows to the render) and passes `viewedGroup={viewedGroup}`. The `__mocks__` copy and its test render with `viewedGroup={{ id: 1, slug: 'alpha' }}`.
- **The 9 `PageComponent` implementations** (`HomePageContent`, `PayOffPageContent`, `ListAllSessions`, `BirdSummary`, `HighlightsPageContent`, `SpPage`, `SppStatsTable`, `ControlsPage`, `RingSequencesPage`) take `viewedGroup: ViewedGroup` and read `viewedGroup.id` where they previously used the plain id. Grandchildren (`SpStats`, `SessionHistoryCalendar`, `RingSizeSection`, …) keep receiving a plain `viewedGroupId: number` — the object shape does not cascade past the boundary.
- **The 10 `/group/[groupId]/**` wrapper files** drop the now-redundant `viewedGroupId={viewedGroupId}` prop, passing only `viewedGroup`.
- **The route-page override prop types** drop `viewedGroupId?: number` entirely — `viewedGroup` is the sole override input.

## Notes / assumptions (subagent mode)

- Used the shared `ViewedGroup` type (from `@/lib/group-slug`) for the new prop rather than re-spelling the inline `{ id: number; slug: string | null }` literal each time — it is exactly that shape, and the route pages already import it (DRY / consistency per CLAUDE.md). Client components import it type-only, so it is erased from the client bundle.
- `SessionSummary` (the session route's `PageComponent`) reads `viewedGroupId` from `params`, not a top-level prop, so it stays assignable to the boundary unchanged and needs no edit — consistent with the ticket excluding it.

## Tests

- `BootstrapPageData` test's `TestPageComponent` consumes `viewedGroup`; the two boundary-component tests (`SppStatsTable`, `RingSequencesPage`) render with `viewedGroup={{ id, slug }}`; the two `/group` wrapper tests drop the stale `viewedGroupId` assertion.
- Full app suite green (738 passing), `type:check` clean, lint 0 errors, E2E safe suite (91) green via the pre-push hook.

Closes #488.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by wheresrhys** [comment]
## Behaviour change: `viewedGroup` object replaces `viewedGroupId` at the PageComponent boundary — and stops there

The value crossing the `BootstrapPageData → PageComponent` boundary changes shape from a bare number to a `{ id, slug }` object. The subtle part is **where the new shape stops**: it does *not* cascade — grandchildren below each PageComponent keep receiving a plain `viewedGroupId: number`, exactly as before. The `/group` wrapper layer also stops sending its duplicate `viewedGroupId` prop.

Worked example: viewing group `2` (slug `alpha`) on the species page.

```mermaid
flowchart TB
    subgraph OLD["Before"]
        direction TB
        w1["/group/[2] wrapper"] -->|"viewedGroupId=2<br/>viewedGroup={id:2,slug}"| r1["route page (SpeciesPage)"]
        r1 -->|"viewedGroup"| b1["BootstrapPageData"]
        b1 -->|"viewedGroupId=2"| p1["PageComponent (SpPage)"]
        p1 -->|"viewedGroupId=2"| g1["grandchildren<br/>(SpStats, tabs…)"]
    end
    subgraph NEW["After"]
        direction TB
        w2["/group/[2] wrapper"] -->|"viewedGroup={id:2,slug}<br/>(redundant id prop dropped)"| r2["route page (SpeciesPage)"]
        r2 -->|"viewedGroup"| b2["BootstrapPageData"]
        b2 -->|"viewedGroup={id:2,slug}"| p2["PageComponent (SpPage)"]
        p2 -->|"viewedGroupId = viewedGroup.id = 2"| g2["grandchildren<br/>(SpStats, tabs…)<br/>UNCHANGED: plain number"]
    end
    OLD --> NEW
```

What moved at each layer:

| Layer | Before | After |
|---|---|---|
| `/group/[groupId]/**` wrapper → route page | `viewedGroupId={n}` **and** `viewedGroup={…}` | `viewedGroup={…}` only |
| Route-page override prop type | `viewedGroupId?: number` + `viewedGroup?` | `viewedGroup?` only (sole override) |
| `BootstrapPageData` → `PageComponent` | `viewedGroupId: number` | `viewedGroup: ViewedGroup` |
| `PageComponent` → its children | `viewedGroupId: number` | `viewedGroupId: number` (unchanged — `viewedGroup.id` unwrapped at the boundary) |

Secondary effect inside `LoadWithData`: the empty-group guard now branches on the resolved `viewedGroup` object itself (`if (!viewedGroup)`) rather than on `viewedGroupId`, so the non-null object is what gets handed to the render.

File: app/(routes)/bird/[ring]/page.tsx [hunk 1]
```diff
@@ -77,12 +77,13 @@ export async function fetchBirdData(
...
 	data: bird,
-	viewedGroupId: loggedInGroupId
+	viewedGroup
 }: {
 	params: PageParams;
 	data: StandaloneBird;
-	viewedGroupId: number;
+	viewedGroup: ViewedGroup;
 }) {
+	const loggedInGroupId = viewedGroup.id;
```
Comment (wheresrhys) [inline app/(routes)/bird/[ring]/page.tsx]: can we drop this and just use viewedGroup.id in the filter below

Comment (wheresrhys) [reply app/(routes)/bird/[ring]/page.tsx]: Done in bb434a6 — dropped the `loggedInGroupId` local and now use `viewedGroup.id` directly in the `sharedEncounterCount` filter. Bird page tests + full E2E suite pass.
